clickstream-event-visualiser-ui/src/main/java/clickstream/eventvisualiser/ui/internal/data/datasource/CSEvDatasourceImpl.kt

```diff
@@ -67,43 +81,79 @@ internal class CSEvDatasourceImpl private constructor(private val evObserver: CS
 
     private suspend fun addInterceptedEvent(eventList: List<CSEventModel>) =
         withContext(Dispatchers.Default) {
-            eventList.forEach {
-                when (it) {
-                    is CSEventModel.Instant -> addNewValueInCurrentList(
-                        CSEvEvent(
-                            eventName = it.eventName ?: "Empty event",
-                            eventId = it.eventId,
-                            properties = it.properties,
-                            state = CSEvState.ACKNOWLEDGED,
-                            timeStampInMillis = it.timeStamp * 1000L
-                        )
+            eventList.forEach { csEvent ->
+                when (csEvent) {
+                    is CSEventModel.Event -> {
+                        handleEventData(csEvent)
+                    }
+                    is CSEventModel.Connection -> {
+                        internalConnectionFlow.update {
+                            csEvent.isConnected
+                        }
+                    }
+                    else -> {
+                        // No-op
+                    }
+                }
+            }
+        }
+
+    private fun handleEventData(csEvent: CSEventModel.Event) {
+        when (csEvent) {
+            is CSEventModel.Event.Instant -> addNewValueInCurrentList(
+                CSEvEvent(
```

Comment:
the construction for CSEvEvent also duplicated with `CSEventModel.Event.Scheduled`, can we have an extract function for this?